### ISSUE TITLE
[gateway-gcs] Filter out ZZZZ-Minio/

### ISSUE THREAD

**Issue by krishnasrinivas** [minio/minio#4415]
Code :
```
341                                                                                                                                                                                     
342                 if attrs.Prefix == ZZZZMinioPrefix {                                                                                                                                
343                         // we don't return our metadata prefix                                                                                                                      
344                         continue                                                                                                                                                    
345                 } else if attrs.Prefix != "" {                                                                                                                                      
346                         prefixes = append(prefixes, attrs.Prefix)                                                                                                                   
347                         continue                                                                                                                                                    
348                 
```

But `mc ls` lists:

```
krishna@escape:~$ mc ls myminio/krisbuck
[2017-05-23 12:21:19 PDT] 2.2KiB 1
[2017-05-23 12:21:37 PDT] 2.2KiB 10
[2017-05-23 12:21:39 PDT] 2.2KiB 11
[2017-05-23 12:21:41 PDT] 2.2KiB 12
[2017-05-23 12:21:42 PDT] 2.2KiB 13
[2017-05-23 12:21:44 PDT] 2.2KiB 14
[2017-05-23 12:21:46 PDT] 2.2KiB 15
[2017-05-23 12:21:48 PDT] 2.2KiB 16
[2017-05-23 12:21:50 PDT] 2.2KiB 17
[2017-05-23 12:21:52 PDT] 2.2KiB 18
[2017-05-23 12:21:54 PDT] 2.2KiB 19
[2017-05-23 12:21:21 PDT] 2.2KiB 2
[2017-05-23 12:21:56 PDT] 2.2KiB 20
[2017-05-23 12:21:23 PDT] 2.2KiB 3
[2017-05-23 12:21:25 PDT] 2.2KiB 4
[2017-05-23 12:21:27 PDT] 2.2KiB 5
[2017-05-23 12:21:29 PDT] 2.2KiB 6
[2017-05-23 12:21:31 PDT] 2.2KiB 7
[2017-05-23 12:21:33 PDT] 2.2KiB 8
[2017-05-23 12:21:35 PDT] 2.2KiB 9
[2017-05-23 16:25:45 PDT]     0B ZZZZ-Minio/

```

**Comment by krisis** [comment]
@krishnasrinivas, please fill out our github PR template. It helps us to gather all the information relevant to the issue and provides better context. This issue doesn't convey much. E.g, I can't say if I am facing this issue as a user.

**Comment by krisis** [comment]
More information from slack conversation with @krishnasrinivas,
>yes you can fix the gcs code to filter it out, the desired behavior is also that we should not show our "metadata" in the list output, this is only for internal consumption

**Comment by krisis** [comment]
## Steps to reproduce
1. `mc cp /etc/lsb-release mygcs/krisis/with-prefix/obj1`, where `krisis` is the bucket.
2. `mc ls mygcs/krisis`

## Current output
```
➜  ~ mc ls s1/krisis/             
[2017-05-23 19:07:20 IST]   105B anon-upload
[2017-05-23 18:40:39 IST]   105B lsb-release
[2017-05-23 19:13:56 IST]   105B obj
[2017-05-23 19:13:59 IST]   105B obj-copy
[2017-05-24 11:46:59 IST]     0B ZZZZ-Minio/
[2017-05-24 11:46:59 IST]     0B with-prefix/
```

## Expected output
```
➜  ~ mc ls s1/krisis/             
[2017-05-23 19:07:20 IST]   105B anon-upload
[2017-05-23 18:40:39 IST]   105B lsb-release
[2017-05-23 19:13:56 IST]   105B obj
[2017-05-23 19:13:59 IST]   105B obj-copy
[2017-05-24 11:46:59 IST]     0B with-prefix/
```

**Comment by krisis** [comment]
Following is a functional test to reproduce the issue,
```go
package main

import (
	"encoding/json"
	"fmt"
	"log"

	minio "github.com/minio/minio-go"
)

func main() {
	clnt, err := minio.NewCore("localhost:9000", "minio", "minio123", false)
	if err != nil {
		log.Fatalln(err)
	}

	bucket := "krisis"
	res, err := clnt.ListObjects(bucket, "", "", "/", 1000)
	if err != nil {
		log.Fatalln(err)
	}

	if len(res.CommonPrefixes) != 1 {
		log.Fatalln("Expected only one prefix but have more %v", res.CommonPrefixes)
	}

	jsonBytes, err := json.Marshal(res)
	if err != nil {
		log.Fatalln(err)
	}
	fmt.Println(string(jsonBytes))
}

```

**Comment by krisis** [comment]
@harshavardhana this issue is fixed by https://github.com/minio/minio/pull/4435, can be closed.

**Comment by harshavardhana** [comment]
> @harshavardhana this issue is fixed by #4435, can be closed.

Fixed in #4435 - closing.
